docs/source/distribution_implementation.md

```diff
@@ -0,0 +1,452 @@
+# Implementing a Distribution (devs)
+
+This guide provides an overview on how to implement a distribution for version 4
+of PyMC3. It is designed for developers who wish to add a new distribution to
+the library. Users will not be aware of all this complexity and should instead
+make use of helper methods such as (TODO?)
+
+PyMC3 distributions build on top of Aesara's random variables (TODO: add link),
+and implement `logp` and `logcdf` methods as well as other initialization and
+validation helpers, most notably `shape/dims`, alternative parametrizations, and
+default `transforms`.
+
+Here is a summary check-list of the steps needed to implement a new distribution.
+Each section will be expanded below:
+
+1. Creating a new `RandomVariable` `op`
+1. Implementing the corresponding `Distribution` class
+1. Adding tests for the new `RandomVariable`
+1. Adding tests for the `logp` / `logcdf` methods
+1. Documenting the new `Distribution`.
+
+This guide does not attempt to explain the rationale behind the `Distributions`
+current implementation, and details are provided only insofar as they help
+implementing new "standard" distributions.
```

Comment:
>  This guide does not attempt to explain the rationale behind the `Distributions` current implementation

Would you say that [this PR](https://github.com/pymc-devs/pymc3/pull/4440) is a good reference for anyone interested in that? if so, would it make sense to add it here for anyone interested in better understanding the ratio behind the change in implementation?

Reply:
After having a look, I feel it's too unstructured to be of real help. Besides it was not even merged in the end. This one contains all the changes, but it's gigantic: https://github.com/pymc-devs/pymc3/pull/4500